app/models/event_schedule.rb

```diff
@@ -37,6 +39,16 @@ def replacement?
 
   private
 
+  def start_after_end_hour
+    return unless errors.blank? && event.program.conference.try(:end_hour)
```

Comment:
ok, to be honest I am not sure if this is better than to check if they are present as you did before. Maybe using `try` for everything like `event.try(:program).try(:conference).try(:end_hour)`.

To be honest I don't have a strong opinion here, so write it with `errors`, `try` or checking if they are present as you had done before, as you think is better. But what about at least moving this to the validation definition? I mean:

`validate :start_after_end_hour if event.try(:program).try(:conference).try(:end_hour)`
`validate :start_before_start_hour if event.try(:program).try(:conference).try(:start_hour)`

Because really when the time is not present you can not validate it. :thinking: